### PR TITLE
[BUGFIX][shapeinference]: Fixed a bug in get_input_bounds with op with no input on specifed port

### PR DESCRIPTION
**Pull request by pkowalc1**
### Details:
Fixed a bug in get_input_bounds(), where method tries to access input at specified port, but operator does not have anything attached at that port - the result was a crash. 

This situation was encountered when calling shape_infer() method with slice op, which did not have any attached inputs.

### CODE REVIEW

File: src/core/tests/utils_tests.cpp [hunk 1]
```diff
@@ -0,0 +1,17 @@
+// Copyright (C) 2018-2023 Intel Corporation
```
Comment (praasz) [inline src/core/tests/utils_tests.cpp:1]:
```suggestion
// Copyright (C) 2018-2024 Intel Corporation
```

File: src/core/tests/utils_tests.cpp [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "../shape_inference/include/tensor_data_accessor.hpp"
+#include "../shape_inference/include/utils.hpp"
```
Comment (praasz) [inline src/core/tests/utils_tests.cpp:6]:
```suggestion
#include "utils.hpp"
```

File: src/core/tests/utils_tests.cpp [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+#include "gtest/gtest.h"
```
Comment (praasz) [inline src/core/tests/utils_tests.cpp:7]:
```suggestion
#include <gtest/gtest.h>
```

File: src/core/tests/utils_tests.cpp
Comment (praasz) [inline src/core/tests/utils_tests.cpp:1]: Rename `shape_inference_utils_test.cpp` to be more dedicated.

File: src/core/tests/utils_tests.cpp (hunk 1 above)
Comment (pkowalc1) [reply src/core/tests/utils_tests.cpp:1]: Fixed.

File: src/core/tests/utils_tests.cpp (hunk 2 above)
Comment (pkowalc1) [reply src/core/tests/utils_tests.cpp:6]: Fixed.

File: src/core/tests/utils_tests.cpp
Comment (pkowalc1) [reply src/core/tests/utils_tests.cpp:1]: Fixed.

**Review by praasz — APPROVED** [review]

**Review by vurusovs — APPROVED** [review]

**Pull request merged**